File: apps/s-2-admin/src/api/eventTypeCustomInput/EventTypeCustomInput.ts
import { EventType } from "../eventType/EventType";

export type EventTypeCustomInput = {
  id: number;
  eventType?: EventType;
  label: string;
  type?: "TEXT" | "TEXTLONG" | "NUMBER" | "BOOL";
  required: boolean;
  placeholder: string;
};

File: apps/cactus-demo-dotnet/src/Infrastructure/Models/EventTypeCustomInput.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CactusDemoDotnet.Core.Enums;

namespace CactusDemoDotnet.Infrastructure.Models;

[Table("EventTypeCustomInputs")]
public class EventTypeCustomInputDbModel
{
    [Key()]
    [Required()]
    public int Id { get; set; }

    public int EventTypeId { get; set; }

    [ForeignKey(nameof(EventTypeId))]
    public EventTypeDbModel EventType { get; set; } = null;

    [Required()]
    [StringLength(256)]
    public string Label { get; set; }

    [Required()]
    public TypeEnum Type { get; set; } = TypeEnum.TEXT;

    [Required()]
    public bool Required { get; set; }

    [Required()]
    [StringLength(256)]
    public string Placeholder { get; set; }
}
